packages/ERTP/src/issuer.chainmail

```diff
@@ -0,0 +1,213 @@
+ * allegedName doesn't have to be a string, but it will only be used for
+ * its value. The allegedName is useful for debugging and double-checking
+ * assumptions, but should not be trusted. 
+ * 
+ * The mathHelpersName will be used to import a specific mathHelpers
+ * from the mathHelpers library. For example, natMathHelpers, the
+ * default, is used for basic fungible tokens.
+ */
+interface MintMaker {
+  produceIssuer(
+      allegedName :Comparable,
+      mathHelperName :String) -> (IssuerObjs);
+}
+
+/**
+ * The return value of produceIssuer
+ */
+struct IssuerObjs ( ) {
+  mint :Mint;
+  issuer :Issuer;
+  amountMath :AmountMath;
+  brand :Brand;
+}
+
+/**
+ * Holding a Mint carries the right to issue new digital assets. These
+ * assets all have the same kind, which is called a Brand.
+ */
+interface Mint (Amount (Extent)) {
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /**
+   * Create a new Payment containing newly minted amount. 
+   */
+  mintPayment(newAmount :Amount, memo :String = 'payment') -> (Payment);
+}
+
+/**
+ * Purses hold amount of digital assets of the same brand, but unlike Payments, they are
+ * not meant to be sent to others. To transfer digital assets, a
+ * Payment should be withdrawn from a Purse. A purse's balance can rise
+ * and fall, through the action of deposit() and withdraw().
+ *
+ * The primary use for Purses and Payments is for currency-like and goods-like
+ * digital assets, but they can also be used to represent other kinds of rights, such
+ * as the right to participate in a particular contract.
+ */
+interface Purse (Amount) {
+  /** Get the memo field of this purse. */
+  memo() -> (String);
+
+  /** Get the Issuer for this mint. */
+  getIssuer() -> (Issuer);
+
+  /** Get the amount contained in this purse, confirmed by the issuer. */
+  getBalance() -> (Amount);
+
+  /**
+   * Deposit all the contents of srcPayment into this purse, returning the
+   * amount. If the optional argument amount does not equal the balance of srcPayment,
+   * throw error.
+   */
+  deposit(srcPayment :Payment, amount :Amount) -> (Amount);
+
+  /** Withdraw amount from this purse into a new Payment. */
+  withdraw(amount :Amount, name :String) -> (Payment);
+
+}
+
+/**
+ * Payments hold amount of digital assets of the same brand in transit. Payments can
+ * be deposited in purses, split into multiple payments, combined, and
+ * claimed (getting an exclusive payment). Payments are linear, meaning
+ * that either a payment has its full original balance, or it is used up
+ * entirely. It is impossible to partially use a payment. 
+ *
+ * Payments are often received from other actors and therefore should
+ * not be trusted themselves. To get the balance of a payment, use the
+ * trusted issuer: issuer.getBalance(payment),
+ *
+ * Payments can be converted to Purses by getting a verified issuer and calling
+ * issuer.makeEmptyPurse().deposit(payment);
```

Comment:
`.deposit` returns the amount, not the purse, so i feel this one-liner is a bit confusing.

Alternative proposal:
> Payments can be converted to Purses by getting a verified issuer and calling `issuer.makeEmptyPurse();` to create a purse, then `purse.deposit(payment);`

Reply:
I like that better, thanks!